generate_spec_sources.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+
+import sys
+import io
+import yaml
+
+
+COLLECTION_RELEASE = "collection_release.yml"
+SPECFILE_IN = sys.argv[1]
+SPECFILE_OUT = sys.argv[2]
+
+BEGIN_SOURCES_MARKER = "# BEGIN AUTOGENERATED SOURCES"
+END_SOURCES_MARKER = "# END AUTOGENERATED SOURCES"
+
+BEGIN_SETUP_MARKER = "# BEGIN AUTOGENERATED SETUP"
+END_SETUP_MARKER = "# END AUTOGENERATED SETUP"
+
+BEGIN_SETUP_TMPL = "%setup -q "
+END_SETUP_TMPL = " -n %{getarchivedir 0}"
+
+
+class Replacement:
+    """Class describing a replacement of lines between markers in a file"""
+
+    def __init__(self, begin_marker, end_marker, generate):
+        self.begin_marker = begin_marker
+        self.end_marker = end_marker
+        self.generate = generate
+        self.throwaway = False
+        self.marker_seen = False
+
+    def process_line(self, line):
+        # did we change the input line? If not, we are merely returning it.
+        changed = False
+        of = io.StringIO()
+        if not self.throwaway:
+            of.write(line)
+            if line.strip() == self.begin_marker:
+                if self.marker_seen:
+                    sys.exit("Duplicate replacement section")
+                changed = True
+                self.generate(of)
+                self.throwaway = True
+                self.marker_seen = True
+            elif line.strip() == self.end_marker:
+                sys.exit("End marker '{}' without begin marker".format(self.end_marker))
+        else:
+            if line.strip() == self.end_marker:
+                self.throwaway = False
+                of.write(line)
+            elif line.strip() == self.begin_marker:
+                sys.exit("Duplicate begin marker: {}".format(self.begin_marker))
+            else:
+                # we discarded the input line
+                changed = True
+
+        return (changed, of.getvalue())
+
+    def inside(self):
+        """Are we currently inside the replacement block?"""
+        return self.throwaway
+
+    def after(self):
+        """Have we been inside the replacement block?"""
+        return self.marker_seen and not self.throwaway
+
+
+with open(COLLECTION_RELEASE, "r") as cf:
+    roles = yaml.safe_load(cf)
+    for rolename, roleinfo in roles.items():
+        print(roleinfo)
+    rolesbysourcenum = {
+        roleinfo["sourcenum"]: rolename for rolename, roleinfo in roles.items()
+    }
+
+
+def generate_sources(output):
+    k = list(rolesbysourcenum.keys())
+    k.sort()
+
+    for i in k:
+        r = rolesbysourcenum[i]
+        org = roles[rolesbysourcenum[i]].get("org")
```

Comment:
```suggestion
        org = roles[r].get("org")
```
unless semantically you want to differentiate between role `r` and `rolesbysourcenum[i]`
also below